tests/common/snappi/snappi_fixtures.py

```diff
@@ -517,3 +521,355 @@ def cvg_api(snappi_api_serv_ip,
+        port_config_list (list): list of Snappi port configuration information
+        duthost (object): device under test
+        snappi_ports (list): list of Snappi port information
+    Returns:
+        True if we successfully generate configuration or False
+    """
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    if 'minigraph_vlans' in mg_facts:
+        vlan_facts = mg_facts['minigraph_vlans']
+    else:
+        return True
+
+    if len(vlan_facts) == 0:
+        return True
+
+    vlan_member = {}
+    for k, v in vlan_facts.items():
+        vlan_member[k] = v['members']
+
+    vlan_intf_facts = mg_facts['minigraph_vlan_interfaces']
+    vlan_intf = {}
+    for v in vlan_intf_facts:
+        if __valid_ipv4_addr(v['addr']):
+            vlan_intf[v['attachto']] = v
+
+    dut_mac = str(duthost.facts['router_mac'])
+
+    """ For each Vlan """
+    for vlan in vlan_member:
+        phy_intfs = vlan_member[vlan]
+        gw_addr = str(vlan_intf[vlan]['addr'])
+        prefix = str(vlan_intf[vlan]['prefixlen'])
+        vlan_subnet = '{}/{}'.format(gw_addr, prefix)
+        vlan_ip_addrs = get_addrs_in_subnet(vlan_subnet, len(phy_intfs))
+
+        """ For each physical interface attached to this Vlan """
+        for i in range(len(phy_intfs)):
+            phy_intf = phy_intfs[i]
+            vlan_ip_addr = vlan_ip_addrs[i]
+
+            port_ids = [id for id, snappi_port in enumerate(snappi_ports)
+                        if snappi_port['peer_port'] == phy_intf]
+            if len(port_ids) != 1:
+                return False
+
+            port_id = port_ids[0]
+            mac = __gen_mac(port_id)
+            device = config.devices.device(
+                name='Device Port {}'.format(port_id))[-1]
+
+            ethernet = device.ethernets.add()
+            ethernet.name = 'Ethernet Port {}'.format(port_id)
+            ethernet.port_name = config.ports[port_id].name
+            ethernet.mac = mac
+
+            ip_stack = ethernet.ipv4_addresses.add()
+            ip_stack.name = 'Ipv4 Port {}'.format(port_id)
+            ip_stack.address = vlan_ip_addr
+            ip_stack.prefix = int(prefix)
+            ip_stack.gateway = gw_addr
+
+            port_config = SnappiPortConfig(id=port_id,
+                                           ip=vlan_ip_addr,
+                                           mac=mac,
+                                           gw=gw_addr,
+                                           gw_mac=dut_mac,
+                                           prefix_len=prefix,
+                                           port_type=SnappiPortType.VlanMember,
+                                           peer_port=phy_intf)
+
+            port_config_list.append(port_config)
+
+    return True
+
+
+def __intf_config_multidut(config, port_config_list, duthost, snappi_ports):
+    prefix = 24
+    dutIps = create_ip_list("20.0.1.1", len(snappi_ports), mask=prefix)
```

Comment:
Any reasons to use these IP addresses?

Reply:
It seems to me that Snappi port IP addresses depend on the SONiC switch configuration. We cannot assign arbitrary IP addresses here.

Reply:
This comment has not been resolved